### ISSUE TITLE
new `http` is swallowing errors

### ISSUE THREAD

**Issue by mattdesl** [substack/node-browserify#1371]
Errors are being swallowed with the new `http` module. Maybe something to do with fetch and promises? :confused: 

Test case: 

```js
var simpleGet = require('simple-get')
var url = require('url')

simpleGet({
  url: '/assets/data.csv'
}, function (err, res) {
  if (err) throw err
  response  // <-- ReferenceError
    .on('data', function (data) {
      console.log(data)
    })
})
```

With browserify@9.x you get a proper `Uncaught ReferenceError` in console.
With browserify@11.x you get no warning or errors.

**Comment by mattdesl** [comment]
ping @jhiesey since it may be a bug upstream in `stream-http` ?

**Comment by jhiesey** [comment]
Whoops, I missed this when I was gone on vacation. This definitely could be a `stream-http` bug. I'll take a look.

**Comment by jhiesey** [comment]
This is fixed here: https://github.com/jhiesey/stream-http/commit/3bea2a493fb25ee9b2647d236ebc8122d3b7a89f

This fix will be released in `stream-http@2.0.2`

**Comment by feross** [comment]
Nice, okay. Closing this issue now.
